src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -377,6 +387,117 @@ public void WriteValue(string value)
             this.FlushIfBufferThresholdReached();
         }
 
+        private void WriteStringValueInChunks(ReadOnlySpan<char> value)
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer(); 
+            this.bufferWriter.Write(doubleQuote.Slice(0, 1).Span);
+            this.Flush();
+
+            int chunkSize = 2048;

```

Comment:
I was wondering what a good chunk size.

With 2048 I think we're less likely to resize the output buffer even if chunks need escaping, but we'd be renting temporary escaping buffer more frequently. But we may be renting the same escape buffer multiple times and not impact memory allocations. With 4096 we have fewer chunks, and therefore fewer times we'd have to go through the escape path if some chunks need escaping. 

Since we don't necessarily flush after every chunk, I think 2048 is probably fine.

Reply:
I think having benchmarks on this will guide this decision.